components/automate-deployment/pkg/backup/artifact_streams.go

```diff
@@ -0,0 +1,449 @@
+package backup
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"strings"
+
+	"go.uber.org/multierr"
+)
+
+// ArtifactStream streams a list of artifacts keys
+// by name. A stream may not produce the same value
+// twice.
+// Returns io.EOF when no more elements will be produced
```

Comment:
Might be good to document either here or on the implementations around what our expectations are around copying of these stream structs and thread-safety.  From reading the implementation, it appears that these aren't concurrency-safe (which is fine) so we might want to document that.

Reply:
> A stream may not produce the same value twice.

I think another assumption is that a stream produces sorted values?

Reply:
yep, both true